### PR TITLE
Fix JSON error when comment has no reactions

### PR DESCRIPTION
**Pull request by nickvergessen**
Otherwise talk integration tests spam this error:
```json
{
  "Exception": "Error",
  "Message": "json_decode(): Passing null to parameter #1 ($json) of type string is deprecated at /home/nickv/Nextcloud/25/server/lib/private/Comments/Manager.php#113",
  "Code": 0,
  "Trace": [
    {
      "function": "onAll",
      "class": "OC\\Log\\ErrorHandler",
      "type": "::"
    },
    {
      "file": "/home/nickv/Nextcloud/25/server/lib/private/Comments/Manager.php",
      "line": 113,
      "function": "json_decode"
    },
    {
      "file": "/home/nickv/Nextcloud/25/server/appsbabies/spreed/lib/Chat/CommentsManager.php",
      "line": 40,
      "function": "normalizeDatabaseData",
      "class": "OC\\Comments\\Manager",
      "type": "->"
    },
    {
      "file": "/home/nickv/Nextcloud/25/server/lib/private/Comments/Manager.php",
      "line": 299,
      "function": "getCommentFromData",
      "class": "OCA\\Talk\\Chat\\CommentsManager",
      "type": "->"
    },
    {
      "file": "/home/nickv/Nextcloud/25/server/appsbabies/spreed/lib/Manager.php",
      "line": 227,
      "function": "get",
      "class": "OC\\Comments\\Manager",
      "type": "->"
    },
    {
      "file": "/home/nickv/Nextcloud/25/server/appsbabies/spreed/lib/Room.php",
      "line": 351,
      "function": "loadLastCommentInfo",
      "class": "OCA\\Talk\\Manager",
      "type": "->"
    },
    {
      "file": "/home/nickv/Nextcloud/25/server/appsbabies/spreed/lib/Controller/RoomController.php",
      "line": 534,
      "function": "getLastMessage",
      "class": "OCA\\Talk\\Room",
      "type": "->"
    },
    {
      "file": "/home/nickv/Nextcloud/25/server/appsbabies/spreed/lib/Controller/RoomController.php",
      "line": 346,
      "function": "formatRoomV4",
      "class": "OCA\\Talk\\Controller\\RoomController",
      "type": "->"
    },
    {
      "file": "/home/nickv/Nextcloud/25/server/appsbabies/spreed/lib/Controller/RoomController.php",
      "line": 1395,
      "function": "formatRoom",
      "class": "OCA\\Talk\\Controller\\RoomController",
      "type": "->"
    },
    {
      "file": "/home/nickv/Nextcloud/25/server/lib/private/AppFramework/Http/Dispatcher.php",
      "line": 225,
      "function": "joinRoom",
      "class": "OCA\\Talk\\Controller\\RoomController",
      "type": "->"
    },
    {
      "file": "/home/nickv/Nextcloud/25/server/lib/private/AppFramework/Http/Dispatcher.php",
      "line": 133,
      "function": "executeController",
      "class": "OC\\AppFramework\\Http\\Dispatcher",
      "type": "->"
    },
    {
      "file": "/home/nickv/Nextcloud/25/server/lib/private/AppFramework/App.php",
      "line": 172,
      "function": "dispatch",
      "class": "OC\\AppFramework\\Http\\Dispatcher",
      "type": "->"
    },
    {
      "file": "/home/nickv/Nextcloud/25/server/lib/private/Route/Router.php",
      "line": 298,
      "function": "main",
      "class": "OC\\AppFramework\\App",
      "type": "::"
    },
    {
      "file": "/home/nickv/Nextcloud/25/server/ocs/v1.php",
      "line": 62,
      "function": "match",
      "class": "OC\\Route\\Router",
      "type": "->"
    },
    {
      "file": "/home/nickv/Nextcloud/25/server/ocs/v2.php",
      "line": 23,
      "args": [
        "/home/nickv/Nextcloud/25/server/ocs/v1.php"
      ],
      "function": "require_once"
    }
  ],
  "File": "/home/nickv/Nextcloud/25/server/lib/private/Log/ErrorHandler.php",
  "Line": 99,
  "CustomMessage": "--"
}
```

### CODE REVIEW

**Review by blizzz — APPROVED** [review]

File: lib/private/Comments/Manager.php [hunk 1]
```diff
@@ -110,18 +110,22 @@ protected function normalizeDatabaseData(array $data) {
...
-			// Ordering does not work on the database with group concat and Oracle,
-			// So we simply sort on the output.
-			if (is_array($list)) {
-				uasort($list, static function ($a, $b) {
-					if ($a === $b) {
-						return 0;
-					}
-					return ($a > $b) ? -1 : 1;
-				});
+			if ($data['reactions']) {
```
Comment (come-nc) [inline lib/private/Comments/Manager.php:113]: Looks weird, what values (and types) are possible for `$data['reactions']`? Is it always set?
Was expecting an `isset` check, or `is_string`, or `!== null`.

File: lib/private/Comments/Manager.php [hunk 2]
```diff
@@ -110,18 +110,22 @@ protected function normalizeDatabaseData(array $data) {
...
+				// So we simply sort on the output.
+				if (is_array($list)) {
+					uasort($list, static function ($a, $b) {
+						if ($a === $b) {
+							return 0;
+						}
+						return ($a > $b) ? -1 : 1;
+					});
+				}
+				$data['reactions'] = $list;
```
Comment (come-nc) [inline lib/private/Comments/Manager.php:125]: So `$data['reactions']` may not be an array at the end of this function, won’t that cause trouble later?
Please add an else to the is_array check, and either default to an empty array or throw an error in there.

Comment (nickvergessen) [reply lib/private/Comments/Manager.php:125]: It is always an array here. the is_array check was basically the wrong approach from catching json_decode of null.
But that logs an error in 8.1+ it seems.

But we can add more hardening to ensure it

File: lib/private/Comments/Manager.php (hunk 1 above)
Comment (nickvergessen) [reply lib/private/Comments/Manager.php:113]: is_string or !== null both do the job. those are the 2 options

**Review by come-nc — APPROVED** [review]

**Review by vitormattos — APPROVED** [review]

**Pull request merged**